packages/codemods/src/plugins/default.ts

```diff
@@ -1,17 +1,151 @@
-import type { JSXAttribute } from 'jscodeshift';
+import type {
+  CommentBlock,
+  CommentLine,
+  JSXAttribute,
+  ImportDeclaration,
+  ImportSpecifier,
+  ImportDefaultSpecifier,
+  ImportNamespaceSpecifier,
+} from 'jscodeshift';
 
 import type { CodemodPlugin } from './types';
 
+// This is required since the imported ImportDeclaration is incorrectly typed.
+// ? Although improperly typed, it doesn't seem to have any tangible impact on code structure.
+// ? Maybe good to copy over for semantic reasons?
+export interface ImportDeclarationWithExtraProperties extends ImportDeclaration {
+  innerComments?: (CommentLine | CommentBlock)[];
+  leadingComments?: (CommentLine | CommentBlock)[];
+  trailingComments?: (CommentLine | CommentBlock)[];
+}
+
+const isImportSpecifier = (
+  specifier: ImportSpecifier | ImportDefaultSpecifier | ImportNamespaceSpecifier
+): specifier is ImportSpecifier => {
+  return (specifier as ImportSpecifier).imported !== undefined;
+};
+
 const defaultCodemodPlugin: CodemodPlugin = {
   name: 'default-plugin',
   create: (_, { jscodeshift: j }) => ({
     transform: {
       buildImport({ compiledImportPath, currentNode, specifiers }) {
-        const newImport = j.importDeclaration(specifiers, j.literal(compiledImportPath));
+        const newImport: ImportDeclarationWithExtraProperties = j.importDeclaration(
+          specifiers,
+          j.literal(compiledImportPath)
+        );
 
-        // Copy the comments from the previous import to the new one
+        // ! Note: Comments are copied over imperfectly.
+        // This is due to open recast Issue #191 where trailing comments are being turned into leading comments. Link: https://github.com/benjamn/recast/issues/191
+        // Also, jscodeshift does not accept location information for comments.
+        // Expect inline comments to be bumped up one row.
+
+        // Copy the comments from the previous import to the new one.
         newImport.comments = currentNode.comments;
 
+        // Copy over inline comments from the previous import to the new one
+        currentNode.specifiers?.forEach((specifier) => {
```

Comment:
What if we keep the comments when creating the [newSpecifiers](https://github.com/atlassian-labs/compiled/blob/master/packages/codemods/src/utils/main.tsx#L269), would that help to simplify the logic?